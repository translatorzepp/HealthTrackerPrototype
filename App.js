/**
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import {Platform, StyleSheet, Text, TextInput, View} from 'react-native';

type Props = {};

export default class App extends Component<Props> {
  render() {
    return (
      <View>
        <DateBanner styleStuff={styles} today={this.todayString()} />
        <View style={styles.container}>
          <Text style={styles.prompt}>What have you eaten today?</Text>
          <RecordTodaysFood></RecordTodaysFood>
        </View>
      </View>
    );
  }

  todayString() {
    return 'Today, ' + new Date().toDateString();
  }
}

class RecordTodaysFood extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = {
      foodInputs: [""],
    }
  }

  render() {
    let inputs = this.state.foodInputs.map((item, index) => {
      return(
        <FoodNameInput
          foodName={item}
          key={index}
          index={index}

          updateFoodInputs={(text, index) => {
            var newFoodInputs = this.state.foodInputs;
            newFoodInputs[index] = text;

            if (text.length >= 2 && !this._emptyInputAlreadyExists()) {
              newFoodInputs[index + 1] = "";
            }

            this.setState({
              foodInputs: newFoodInputs,
            });
          }}
        />
      );
    })

    return(
      <View>
        {inputs}
      </View>
    );
  }

  _emptyInputAlreadyExists() {
    var anyEmpty = false;
    for (const e of this.state.foodInputs) {
      if (e.length == 0) {
        anyEmpty = true;
        break;
      }
    }
    return anyEmpty;
  }
}

class FoodNameInput extends Component<Props> {
  constructor(props) {
    super(props);

    var empty = this.props.foodName.length <= 2
    this.state = {
      nameOfFood: this.props.foodName,
      foodEntered: !empty,
    }
  }

  _submitFood(text)  {
    var filledOut = text.length > 2
    this.setState({
      foodEntered: filledOut,
      nameOfFood: text
    });

    this.props.updateFoodInputs(text, this.props.index);
  }

  render() {
    var colorForBorder = this.state.foodEntered ? 'seagreen' : 'royalblue';
    var widthForBorder = this.state.foodEntered ? 1 : 2
    return (
      <TextInput
        placeholder="Name of a food"
        style={{borderColor: colorForBorder, borderWidth: widthForBorder}}
        onEndEditing={(event) => this._submitFood(event.nativeEvent.text)}
      >
        {this.state.nameOfFood}
      </TextInput>
    )
  }
}

class DateBanner extends Component<Props> {
  render() {
    return (
      <View style={this.props.styleStuff.dateBackground}>
        <Text style={this.props.styleStuff.dateText}>{this.props.today}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  dateBackground: {
    width: '100%',
    height: '20%',
    backgroundColor: 'seagreen',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: '10%',
  },
  dateText: {
    fontSize: 20,
    textAlign: 'center',
    color: 'seashell',
  },
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  prompt: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: '4%',
  },
});
