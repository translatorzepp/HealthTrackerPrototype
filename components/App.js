/**
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import {Text, View} from 'react-native';
import RecordTodaysFood from '../components/RecordTodaysFood';
import styles from '../styles';

type Props = {};

export default class App extends Component<Props> {
  _todayDate() {
    return new Date();
  }

  render() {
    const todayDate = this._todayDate();
    return (
      <View style={styles.topStyle}>
        <DateBanner today={todayDate} />
        <RecordTodaysFood date={todayDate} startFood={[]}></RecordTodaysFood>
      </View>
    );
  }
}

class DateBanner extends Component<Props> {
  render() {
    return (
      <View style={styles.dateBackground}>
        <Text style={styles.dateText}>{this.todayHeaderDisplay(this.props.today)}</Text>
      </View>
    );
  }

  todayHeaderDisplay(todayDate) {
    return 'Today, ' + todayDate.toDateString();
  }
}
