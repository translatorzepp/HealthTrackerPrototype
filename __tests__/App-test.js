/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MockDate from 'mockdate';
import AsyncStorage from '@react-native-community/async-storage';

beforeAll(() =>  {
  MockDate.set('09/13/2019');
});

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('requests stored items for the day', () => {
  const app = renderer.create(<App />);

  expect(AsyncStorage.getItem).toHaveBeenNthCalledWith(1, '20190913')
  expect(AsyncStorage.getItem).toHaveBeenNthCalledWith(2, '20190913')
});

// it('displays stored items for the day', () => {
//   const tree = renderer.create(<App />).toJSON();
//   // TODO: mock retrieved data
//   expect(tree).toMatchSnapshot();
// });

afterAll(() => {
  MockDate.reset();
});
