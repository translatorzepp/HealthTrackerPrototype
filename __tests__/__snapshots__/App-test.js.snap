// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "seagreen",
        "height": "10%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "seashell",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Today, Fri Sep 13 2019
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "height": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "textAlign": "center",
        }
      }
    >
      What have you eaten today?
    </Text>
    <View
      style={
        Object {
          "marginBottom": "5%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Name of a food"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "royalblue",
            "borderWidth": 2,
          }
        }
        underlineColorAndroid="transparent"
      >
        
      </TextInput>
      <View
        accessibilityLabel="Save Food"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Save Food
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "flex-end",
        }
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={Object {}}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
